### PR TITLE
feat: support like syntax in show tables statement

### LINKED ISSUE #128: Show tables statement support name like syntax
**Description**

Currently, `show tables` is supported by CeresDB, it would be convenient to support name like syntax like in MySQL.
```
SHOW TABLES LIKE 'pattern'
```

**Proposal**

Support this syntax with reference with MySQL

**Additional context**

Currently it will return error below
- https://github.com/CeresDB/ceresdb/blob/334bc6ce13aac2e72a0771419b4aefe2164255e9/tests/cases/01_system/system_tables.result#L15

ref #154

### PR DESCRIPTION
**Pull request by QuintinTao**
### *Which issue does this PR close?*
Close #128
### **Rationale for this change**
no
### **What changes are included in this PR?**

Extend `show tables` statement

### **Are there any user-facing changes?**
user can use show tables like 'abc'  to search table
```
curl --location --request POST 'http://127.0.0.1:5440/sql' \
--header 'Content-Type: application/json' \
--data-binary '{
    "query": "show tables like 'abc%'"
}'
```
### **How does this change test**

UT and integration test.

### CODE REVIEW

**Comment by QuintinTao** [comment]
> @QuintinTao Your PR's description doesn't follow our template, you can refer #330 to update yours, thanks.

OK

File: interpreters/src/show.rs [hunk 1]
```diff
@@ -94,16 +95,25 @@ impl ShowInterpreter {
...
-            .iter()
-            .map(|t| t.name().to_string())
-            .collect::<Vec<_>>();
-
+        let tables_names =  match plan.if_fuzzy  {
+            true  => schema
+                .all_tables()
+                .context(FetchTables)?
+                .iter()
+                .filter(|t| t.name().contains::<&str>(plan.fuzzy_target.as_ref().unwrap().as_ref()))
```
[the remark this answers is not in the document]
Comment (QuintinTao) [reply interpreters/src/show.rs]: OK I will finish it soon!

**Comment by QuintinTao** [comment]
> @QuintinTao A test is required, you can add a Integration test in
> 
> * https://github.com/CeresDB/ceresdb/blob/e2c8faf18dd2ee3a5b87219efe02cff6a7d5f8a4/tests/cases/local/01_system/system_tables.sql#L27

I replaced SHOW TABLES `name` LIKE '01_system_table1';  with SHOW TABLES LIKE '01%';
But it seems SHOW TABLES `name` LIKE '01_system_table1' already running.

File: interpreters/src/show.rs [hunk 2]
```diff
@@ -146,13 +159,19 @@ impl ShowInterpreter {
     }
 }
 
+fn search_str(str: &str, search_re: &str) -> bool {
```
[the remark this answers is not in the document]
Comment (QuintinTao) [reply interpreters/src/show.rs:162]: OK

File: sql/src/ast.rs [hunk 3]
```diff
@@ -96,6 +96,11 @@ pub struct AlterAddColumn {
     pub columns: Vec<ColumnDef>,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct ShowTables {
+    pub fuzzy_target: Option<String>,
```
[the remark this answers is not in the document]
Comment (QuintinTao) [reply sql/src/ast.rs:101]: OK

File: tests/cases/local/01_system/system_tables.result [hunk 4]
```diff
@@ -27,7 +27,7 @@ timestamp,catalog,schema,table_name,engine,
 Timestamp(Timestamp(0)),String(StringBytes(b"ceresdb")),String(StringBytes(b"public")),String(StringBytes(b"01_system_table1")),String(StringBytes(b"Analytic")),
 
 
-SHOW TABLES `name` LIKE '01_system_table1';
 
-Failed to execute query, err: Server(ServerError { code: 400, msg: "failed to parse sql. Caused by: Invalid sql, sql: SHOW TABLES `name` LIKE '01_system_table1';, err:sql parser error: Expected end of statement, found: `name`" })
+SHOW TABLES LIKE '01%';
 
+Tables,
```
[the remark this answers is not in the document]
Comment (QuintinTao) [reply tests/cases/local/01_system/system_tables.result:33]: OK

File: interpreters/src/show.rs [hunk 5]
```diff
@@ -146,13 +159,19 @@ impl ShowInterpreter {
...
+    let regex_str = search_re.replace('_', ".").replace('%', ".*");
+    let re = Regex::new(&regex_str).unwrap();
```
[the remark this answers is not in the document]
Comment (QuintinTao) [reply interpreters/src/show.rs:166]: OK  I will

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Comment by jiacai2050** [comment]
@QuintinTao  Thanks for your hard work. 🚀

**Pull request merged**